sentry/src/db/channel.rs

```diff
@@ -67,3 +69,122 @@ pub async fn insert_channel(
+        let validator = validator.as_ref().map(|validator_id| {
+            serde_json::Value::from_str(&format!(r#"[{{"id": "{}"}}]"#, validator_id))
+                .expect("Not a valid json")
+        });
+        let (where_clauses, params) =
+            channel_list_query_params(creator, validator.as_ref(), valid_until_ge);
+        let total_count_params = (where_clauses.clone(), params.clone());
+
+        let channels = pool
+            .run(move |connection| {
+                async move {
+                    // To understand why we use Order by, see Postgres Documentation: https://www.postgresql.org/docs/8.1/queries-limit.html
+                    let statement = format!("SELECT id, creator, deposit_asset, deposit_amount, valid_until, spec FROM channels WHERE {} ORDER BY spec->>'created' DESC LIMIT {} OFFSET {}", where_clauses.join(" AND "), limit, skip);
+                    match connection.prepare(&statement).await {
+                        Ok(stmt) => {
+                            match connection.query(&stmt, params.as_slice()).await {
+                                Ok(rows) => {
+                                    let channels = rows.iter().map(Channel::from).collect();
+
+                                    Ok((channels, connection))
+                                },
+                                Err(e) => Err((e, connection)),
+                            }
+                        },
+                        Err(e) => Err((e, connection)),
+                    }
+                }
+            })
+            .await?;
+
+        Ok(ListChannels {
+            total_count: list_channels_total_count(
+                &pool,
+                (&total_count_params.0, total_count_params.1),
+            )
+            .await?,
+            channels,
+        })
+    }
+
+    async fn list_channels_total_count<'a>(
+        pool: &DbPool,
+        (where_clauses, params): (&'a [String], Vec<&'a (dyn ToSql + Sync)>),
+    ) -> Result<u64, RunError<bb8_postgres::tokio_postgres::Error>> {
+        pool.run(move |connection| {
+            async move {
+                let statement = format!(
+                    "SELECT COUNT(id)::varchar FROM channels WHERE {}",
+                    where_clauses.join(" AND ")
+                );
+                match connection.prepare(&statement).await {
+                    Ok(stmt) => match connection.query_one(&stmt, params.as_slice()).await {
+                        Ok(row) => Ok((row.get::<_, TotalCount>(0).0, connection)),
+                        Err(e) => Err((e, connection)),
+                    },
+                    Err(e) => Err((e, connection)),
+                }
+            }
+        })
+        .await
+    }
+
+    fn channel_list_query_params<'a>(
+        creator: &'a Option<String>,
+        validator: Option<&'a serde_json::Value>,
+        valid_until_ge: &'a DateTime<Utc>,
+    ) -> (Vec<String>, Vec<&'a (dyn ToSql + Sync)>) {
```

Comment:
We can use HashMap later, this made it easier to finish up the code, but it's an improvement we can do. There might be complications with the parameters when we need them for the prepared statements.